lib/Mongo/MongoCollection.php

```diff
@@ -96,6 +110,19 @@ public function __get($name)
         return $this->db->selectCollection($this->name . '.' . $name);
     }
 
+    public function __set($name, $value)
+    {
+        if ('w' == $name) {
+            $this->w = $value;
+            $this->setWriteConcern($this->w, $this->wtimeout);
+        } elseif ('wtimeout' == $name) {
+            $this->wtimeout = $value;
+            $this->setWriteConcern($this->w, $this->wtimeout);
+        } else {
+            trigger_error('trying to set invalid property', E_ERROR);
```

Comment:
Let me check how the original `MongoCollection` object behaves here, especially when setting a property that corresponds to an existing collection. Since it's such a narrow use case we may also just leave the trigger_error in place here.